core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/aggregation/AggregatorFunction.java

```diff
@@ -92,6 +94,12 @@ private static FunctionResolver count() {
                 arguments -> new CountAggregator(arguments, INTEGER))
             .put(new FunctionSignature(functionName, Collections.singletonList(BOOLEAN)),
                 arguments -> new CountAggregator(arguments, INTEGER))
+            .put(new FunctionSignature(functionName, Collections.singletonList(DATE)),
+                arguments -> new CountAggregator(arguments, INTEGER))
+            .put(new FunctionSignature(functionName, Collections.singletonList(DATETIME)),
+                arguments -> new CountAggregator(arguments, INTEGER))
+            .put(new FunctionSignature(functionName, Collections.singletonList(TIMESTAMP)),
+                arguments -> new CountAggregator(arguments, INTEGER))
```

Comment:
I assume SQL `COUNT()` should work with all types. If so, is it more maintainable to generate function signatures from all core types like this way: https://github.com/opendistro-for-elasticsearch/sql/blob/2aeb80b937788d5a7cb4ccdff21f082064898454/core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/operator/predicate/BinaryPredicateOperator.java#L172